File: 151A/main.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main() {
    long long n, k, l, c, d, p, nl, np;
    cin >> n >> k >> l >> c >> d >> p >> nl >> np;

    long long total_drink = k * l / nl;
    long long total_limes = c * d; 
    long long total_salt = p /np ; 

    long long max_toasts = min(total_drink, min(total_limes, total_salt)) / n;

    cout << max_toasts << endl;

    return 0;
}

File: 151A/main.py
n, k, l, c, d, p, nl, np = map(int, input().split())

total_drink = k * l 
total_limes = c * d 
total_salt = p 

toasts_drink = total_drink // nl
toasts_limes = total_limes // 1
toasts_salt = total_salt // np

max_toasts = min(toasts_drink, toasts_limes, toasts_salt)
toasts_per_friend = max_toasts // n

print(toasts_per_friend)
